src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/antibot/modes/CustomAntiBotMode.kt

```diff
@@ -97,25 +98,26 @@ object CustomAntiBotMode : Choice("Custom"), ModuleAntiBot.IAntiBotMode {
             // TODO: replace with multiple choice
             val materialChoice by enumChoice("Material", ArmorMaterialChoice.LEATHER)
 
-            fun isValid(itemStack: ItemStack): Boolean {
-                val materialEquals = (itemStack.item as? ArmorItem)?.material() == materialChoice.material
-                return when (mode) {
-                    Filter.WHITELIST -> materialEquals
-                    Filter.BLACKLIST -> !materialEquals
+            fun isValid(item: Item): Boolean {
+                with((item as? ArmorItem)?.material() == materialChoice.material) {
+                    return when (mode) {
+                        Filter.WHITELIST -> this
+                        Filter.BLACKLIST -> !this
+                    }
                 }
             }
         }
 
         private val values = arrayOf(
-            ArmorConfigurable("Boots", EquipmentType.BOOTS),
-            ArmorConfigurable("Leggings", EquipmentType.LEGGINGS),
-            ArmorConfigurable("Chestplate", EquipmentType.CHESTPLATE),
             ArmorConfigurable("Helmet", EquipmentType.HELMET),
+            ArmorConfigurable("Chestplate", EquipmentType.CHESTPLATE),
+            ArmorConfigurable("Leggings", EquipmentType.LEGGINGS),
+            ArmorConfigurable("Boots", EquipmentType.BOOTS),
         ).onEach(::tree)
```

Comment:
made in a more familiar order (Helmet -> Chestplate -> Leggings - > Boots

Reply:
This will broken indexed iteration. Because the entity armor list is from bottom to top. You should apply `reverse` to it.

Reply:
fixed